instrumentation/wildfly/src/main/java/com/splunk/opentelemetry/middleware/WildFlyAttributesInstrumentationModule.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.middleware;
+
+import static io.opentelemetry.javaagent.tooling.ClassLoaderMatcher.hasClassesNamed;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.isStatic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+import com.google.auto.service.AutoService;
+import com.splunk.opentelemetry.javaagent.bootstrap.MiddlewareHolder;
+import io.opentelemetry.javaagent.tooling.InstrumentationModule;
+import io.opentelemetry.javaagent.tooling.TypeInstrumentation;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.jboss.as.version.ProductConfig;
+
+@AutoService(InstrumentationModule.class)
+public class WildFlyAttributesInstrumentationModule extends InstrumentationModule {
+
+  public WildFlyAttributesInstrumentationModule() {
+    super("wildfly");
+  }
+
+  @Override
+  public ElementMatcher.Junction<ClassLoader> classLoaderMatcher() {
+    return hasClassesNamed("org.jboss.as.version.ProductConfig");
+  }
+
+  @Override
+  public List<TypeInstrumentation> typeInstrumentations() {
+    return Collections.singletonList(new Instrumentation());
+  }
+
+  public static class Instrumentation implements TypeInstrumentation {
+
+    @Override
+    public ElementMatcher<? super TypeDescription> typeMatcher() {
+      return named("org.jboss.as.version.ProductConfig");
+    }
+
+    @Override
+    public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+      return Collections.singletonMap(
+          isMethod().and(isStatic()).and(named("fromFilesystemSlot")),
```

Comment:
Currently every time this method is called, we try to set middleware attributes, which will result in warning messages from MiddlewareHolder. Have you checked how many times this is called during the startup of the wildfly?

Reply:
Very good point. But why do we write debug log in recurring set? If I make addition check here, that will be 2 CAS instead one of 1.

Reply:
Checking it here is not desired, yes. Initial assumption for this MiddlewareHolder was that there is a possibility to instrument _something_ during server startup, that will happen only once and as early as possible, so that middleware attributes will be set by the time a server will start serving requests. And the log will hint the instrumentation point is not the best. So far there were always some init methods, that would cache server version somewhere for internal needs, e.g. JMX queries etc. Maybe wildfly also has something similar, that will call this method and cache the result for later usage?

Reply:
Fixed